### ISSUE TITLE
Training graph obtained from make_fx module is giving error converting to mlir-module.

### ISSUE THREAD

**Issue by pashu123** [llvm/torch-mlir#832]
I am trying to run https://gist.github.com/pashu123/9cb499308ec1db04e0af99e3a86a9251 which gives the training graph. 
To reproduce the error run `python gist-.py`.  Make sure you have torch-mlir and functorch pip-installed.

The error is:
```
error: unhandled: could not import node:  = prim::Store[name="self"](%self.1) # /home/prashant/torch-mlir/mlir_venv/lib/python3.9/site-packages/torch/utils/_pytree.py:99:17

Traceback (most recent call last):
  File "/home/prashant/sgd.py", line 47, in <module>
    module = torch_mlir.compile(
  File "/home/prashant/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir/__init__.py", line 81, in compile
    mb.import_module(scripted._c, class_annotator)
RuntimeError: see diagnostics
```

The torchscript graph obtained is 
```
graph(%self : __torch__.torch.fx.graph_module.forward,
      %params_1.1 : Tensor,
      %params_2.1 : Tensor,
      %args_1.1 : Tensor):
  %72 : float = prim::Constant[value=-0.01]() # <eval_with_key>.1:20:59
  %45 : bool = prim::Constant[value=1]() # <eval_with_key>.1:13:44
  %22 : bool = prim::Constant[value=0]() # <eval_with_key>.1:8:113
  %97 : Device = prim::Constant[value="cpu"]()
  %14 : NoneType = prim::Constant()
  %18 : int = prim::Constant[value=6]() # <eval_with_key>.1:8:56
  %19 : int = prim::Constant[value=0]() # <eval_with_key>.1:8:68
  %23 : int = prim::Constant[value=1]() # <eval_with_key>.1:8:136
  %27 : int = prim::Constant[value=3]() # <eval_with_key>.1:9:47
  %t.1 : Tensor = aten::t(%params_1.1) # <eval_with_key>.1:5:8
  %addmm.1 : Tensor = aten::addmm(%params_2.1, %args_1.1, %t.1, %23, %23) # <eval_with_key>.1:6:12
  %sum_1.1 : Tensor = aten::sum(%addmm.1, %14) # <eval_with_key>.1:7:12
  %ones_like.1 : Tensor = aten::ones_like(%sum_1.1, %18, %19, %97, %22, %23) # <eval_with_key>.1:8:16
  %28 : int[] = prim::ListConstruct(%27, %27)
  %expand.1 : Tensor = aten::expand(%ones_like.1, %28, %22) # <eval_with_key>.1:9:13
  %t_1.1 : Tensor = aten::t(%expand.1) # <eval_with_key>.1:10:10
  %mm.1 : Tensor = aten::mm(%t_1.1, %args_1.1) # <eval_with_key>.1:11:9
  %t_2.1 : Tensor = aten::t(%mm.1) # <eval_with_key>.1:12:10
  %44 : int[] = prim::ListConstruct(%19)
  %sum_2.1 : Tensor = aten::sum(%expand.1, %44, %45, %14) # <eval_with_key>.1:13:12
  %50 : int[] = prim::ListConstruct(%27)
  %view.1 : Tensor = aten::view(%sum_2.1, %50) # <eval_with_key>.1:14:11
  %detach.1 : Tensor = aten::detach(%view.1) # <eval_with_key>.1:15:13
  %detach_1.1 : Tensor = aten::detach(%detach.1) # <eval_with_key>.1:16:15
  %t_3.1 : Tensor = aten::t(%t_2.1) # <eval_with_key>.1:17:10
  %detach_2.1 : Tensor = aten::detach(%t_3.1) # <eval_with_key>.1:18:15
  %detach_3.1 : Tensor = aten::detach(%detach_2.1) # <eval_with_key>.1:19:15
  %add_.1 : Tensor = aten::add_(%params_2.1, %detach_1.1, %72) # <eval_with_key>.1:20:11
  %add__1.1 : Tensor = aten::add_(%params_1.1, %detach_3.1, %72) # <eval_with_key>.1:21:13
  %87 : Tensor[] = prim::ListConstruct(%add__1.1, %add_.1)
  return (%87)
```

**Comment by pashu123** [comment]
@silvasean Any idea why this would be failing?

**Comment by silvasean** [comment]
It looks like the issue is with importing `prim::Store`, but I don't see it in your IR snippet. Can you find out where the prim::Store is coming from and add support for it in the importer?

**Comment by pashu123** [comment]
Okay.

**Comment by pashu123** [comment]
I tried printing the function names here:  https://github.com/llvm/torch-mlir/blob/52c9987115b39c2f6673315a21f6cdff0d1f51ca/python/torch_mlir/dialects/torch/importer/jit_ir/csrc/ivalue_importer.cpp#L496 and seems like if I do something like 
```
if (function->qualname().qualifiedName().find("._pytree") != std::string::npos)
  continue;
```
It goes through.

**Comment by pashu123** [comment]
Should I add the support for `prim::store` or there is something else that needs to be done - remove unnecessary functions from the python session?

**Comment by silvasean** [comment]
In general it is hard to determine which functions are needed and so we don't do it during import -- we rely on SymbolDCE once we are inside MLIR to do that. I would recommend we just add support for prim::Store.

**Comment by pashu123** [comment]
`torch.jit.save()` and `torch.jit.load()` solves the issue. But I will add the support for prim::Store for future reference.

**Comment by ashay** [comment]
Hi @pashu123, I ran into a similar issue.  Before I start working on this, I wanted to check if you're working on this already.  If not, I'm happy to take this up.  I just wanted to make sure I don't step on your toes.  Thanks!

**Comment by pashu123** [comment]
@ashay I am not working on that, you can go ahead!

**Comment by silvasean** [comment]
Fixed as part of #907
